src/controllers/organisationJobs.ts

```diff
@@ -0,0 +1,161 @@
+        });
+    });
+
+    /**
+     * @desc    Update single organisation job
+     * @route   PUT /api/organisation-jobs/:organisationJobId
+     * @access  Admin/Owner
+     */
+    updateOrganisationJob = asyncHandler(async (req, res, next) => {
+        // check if job exists and is not soft deleted
+        const job = await this.organisationJobsRepository.getOrganisationJobById(req.params.organisationJobId);
+
+        // check if organisation exists and user is organisation owner
+        const isOrganisationOwner = await checkOrganisationOwner(req.user.user_id, job.organisation_id);
+
+        // Unauthorised if neither admin nor organisation owner
+        if (!(req.user.role === 'admin' || isOrganisationOwner)) {
+            return next(new ErrorResponse(`Not authorised to create jobs for this listing`, 403));
+        }
+
+        const { job_title, job_description } = req.body;
+
+        const data = {
+            organisation_job_title: job_title,
+            organisation_job_description: job_description,
+        };
+
+        cleanseData(data);
+
+        const rows = await this.organisationJobsRepository.updateOrganisationJobById(data, req.params.organisationJobId);
+
+        res.status(200).json({
+            success: true,
+            data: rows,
+        });
+    });
+
+    /**
+     * @desc    Deactivate (Soft delete) single organisation job
+     * @route   PUT /api/organisation-jobs/:organisationJobId/deactivate
+     * @access  Admin/Owner
+     */
+    deactivateOrganisationJob = asyncHandler(async (req, res, next) => {
+        // check if job exists and not soft deleted
+        const job = await this.organisationJobsRepository.getOrganisationJobById(req.params.organisationJobId);
+
+        // check if organisation exists and user is organisation owner
+        const isOrganisationOwner = await checkOrganisationOwner(req.user.user_id, job.organisation_id);
+
+        // Unauthorised if neither admin nor organisation owner
+        if (!(req.user.role === 'admin' || isOrganisationOwner)) {
+            return next(new ErrorResponse(`Not authorised to create jobs for this listing`, 403));
+        }
+
+        const data = {
+            deleted_on: moment.tz(process.env.DEFAULT_TIMEZONE).toDate(),
+        };
+
+        const rows = await this.organisationJobsRepository.deactivateOrganisationJobById(data, req.params.organisationJobId);
+
+        res.status(200).json({
+            success: true,
+            data: rows,
+        });
+    });
+
+    /**
+     * @desc    Delete single organisation job
+     * @route   DELETE /api/organisation-jobs/:organisationJobId
+     * @access  Admin
+     */
+    deleteOrganisationJob = asyncHandler(async (req, res, next) => {
+        const rows = await this.organisationJobsRepository.deleteOrganisationJobById(req.params.organisationJobId);
+
+        res.status(200).json({
+            success: true,
+            data: rows,
+        });
+    });
+}
```

Comment:
I think we should change the place where we declare the singleton directly in the class file itself. Sorry this is my bad as we did not do this previously. You can see my latest changes on the `jobs.ts` controller and `jobs.route.ts` to understand this

Could you define the singleton here like so:

```suggestion
}

export const organisationJobsController = new OrganisationJobsController(db.organisationJobs, db.organisations);
```